client/my-sites/checkout/composite-checkout/wpcom/components/wp-order-review-line-items.js

```diff
@@ -510,3 +497,18 @@ function SavingsList( { item } ) {
 		</LineItemMeta>
 	);
 }
+
+function DomainDiscountCallout( { item } ) {
+	const translate = useTranslate();
+
+	const isFreeBundledDomainRegistration = item.wpcom_meta?.is_bundled && item.amount.value === 0;
+	if ( isFreeBundledDomainRegistration ) {
+		return <DiscountCalloutUI>{ translate( 'First year free' ) }</DiscountCalloutUI>;
+	}
+
+	const isFreeDomainMapping =
+		item.wpcom_meta?.product_slug === 'domain_map' && item.amount.value === 0;
+	if ( isFreeDomainMapping ) {
+		return <DiscountCalloutUI>{ translate( 'Free with your plan' ) }</DiscountCalloutUI>;
+	}
```

Comment:
I believe you'll need an explicit `return null;` at the end of this because components will report an error if they return `undefined`.

Reply:
Thanks for catching that. I checked again with several kinds of product in the cart to test the different paths:

![image](https://user-images.githubusercontent.com/9310939/84810675-4b070b80-afd1-11ea-9f4a-df3e3d632f9b.png)